datafusion/src/physical_plan/limit.rs

```diff
@@ -300,6 +300,11 @@ impl ExecutionPlan for LocalLimitExec {
             _ => Statistics::default(),
         }
     }
+
+    fn should_repartition_children(&self) -> bool {
+        // No reason to repartition children as this node is just limiting each input partition.
+        false
```

Comment:
This also fixed https://github.com/apache/arrow-datafusion/issues/423 I believe

Reply:
As currently expressed this only prevent repartitioning of direct children, I think you need something more than that for sortedness as discussed on #424